CMakeLists.txt

```diff
@@ -748,6 +748,7 @@ target_sources(
           src/marlin_stubs/M300.cpp
           src/marlin_stubs/M330.cpp
           src/marlin_stubs/M505.cpp
+          src/hw/mainCheck.cpp
```

Comment:
Why is it camelCase when all our files are named in snake_case? Why is it in between of MXXX gcode files? 🙈

Reply:
It is camelCase because all compile time checks are already named somethingCheck.
It is not true that all our files are named in snake_case, see folder src/gui.
As we use PascalCase for classes I consider it strange that file would be named differently than class it contains.

> Why is it in between of MXXX gcode files?

Rebasing put it there.

Renamed to main_check.cpp and moved out of MXXX gcode files.